.github/workflows/benchmark.yml

```diff
@@ -5,21 +5,25 @@ on:
   push:
     branches:
       - "main"
+  schedule:
+    - cron: '0 0 * * *'
 
 jobs:
-
   run_benchmarks:
-    runs-on: ubuntu-latest
+    runs-on: 'ubuntu-22.04'
     steps:
       - name: checkout
         uses: actions/checkout@v3
-      - name: Install python
-        uses: actions/setup-python@v3
-        with:
-          python-version: '3.10'
       - name: Install Requirements
         run: pip install -r requirements.txt
       - name: Run Benchmarks
         run: |-
-          asv machine --yes
-          asv run
+          asv machine --machine ubuntu-22.04 --yes
+          asv run --skip-existing
+      - name: Commit results
+        uses: EndBug/add-and-commit@v9
```

Comment:
Maybe we could do this without a 3rd party action but I think if it works it is OK for now.